domain/consensus/processes/blockvalidator/block_header_in_context.go

```diff
@@ -29,6 +30,15 @@ func (v *blockValidator) ValidateHeaderInContext(blockHash *externalapi.DomainHa
 		if err != nil {
 			return err
 		}
+
+		log.Debug(logger.NewLogClosure(func() string {
+			ghostdagData, err := v.ghostdagDataStore.Get(v.databaseContext, blockHash)
+			if err != nil {
+				panic(err)
```

Comment:
Instead of panicking, populate an error variable outside of the closure and check it afterwards.